src/main/java/io/github/llamarama/team/common/entity/caravantrader/CaravanTraderEntity.java

```diff
@@ -174,4 +176,25 @@ protected SoundEvent getSplashSound() {
         return SoundEvents.ENTITY_GENERIC_SPLASH;
     }
 
+    @Override
+    public void writeCustomDataToNbt(NbtCompound nbt) {
+        super.writeCustomDataToNbt(nbt);
+        nbt.putInt("lifespan", lifespan);
+    }
+
+    @Override
+    public void readCustomDataFromNbt(NbtCompound nbt) {
+        super.readCustomDataFromNbt(nbt);
+        if (nbt.contains("lifespan")) {
```

Comment:
Due to Nbt standards you are supposed to use UpperCamelCase for Nbt keys unless they are Identifiers.
Also I suggest storing it in a public static constant so that you don't get any spelling issues.